client/log_client.go

```diff
@@ -358,10 +359,24 @@ func (c *LogClient) AddSequencedLeaf(ctx context.Context, data []byte, index int
 }
 
 // AddSequencedLeaves adds any number of pre-sequenced leaves to the log.
+// Indexes must be contiguous.
 func (c *LogClient) AddSequencedLeaves(ctx context.Context, dataByIndex map[int64][]byte) error {
+	if len(dataByIndex) == 0 {
+		return nil
+	}
 	leaves := make([]*trillian.LogLeaf, 0, len(dataByIndex))
-	for index, data := range dataByIndex {
-		leaf, err := c.BuildLeaf(data)
+	indexes := make([]int64, 0, len(dataByIndex))
+	for index := range dataByIndex {
+		indexes = append(indexes, index)
+	}
+	sort.Slice(indexes, func(a, b int) bool { return indexes[a] < indexes[b] })
```

Comment:
A bit more efficient algo would be to find `minIndex` and `maxIndex`, and check that `len(dataByIndex) == maxIndex-minIndex` instead of line 376-378?

Reply:
Admittedly a slightly more ugly one because Go doesn't have min/max funcs for `int64`.

Reply:
That might be more efficient, but is it easier to follow?

Reply:
Nope.